client-runtime/src/main/java/com/microsoft/rest/v2/http/NettyClient.java

```diff
@@ -258,31 +264,38 @@ public boolean isDisposed() {
                                 raw.headers().add(header.name(), header.value());
                             }
 
-                            channel.write(raw).addListener(new GenericFutureListener<Future<? super Void>>() {
-                                @Override
-                                public void operationComplete(Future<? super Void> future) throws Exception {
-                                    if (!future.isSuccess()) {
-                                        channelPool.closeAndRelease(channel);
-                                        emitErrorIfSubscribed(future.cause());
+                            try {
+                                channel.write(raw).addListener(new GenericFutureListener<Future<? super Void>>() {
+                                    @Override
+                                    public void operationComplete(Future<? super Void> future) throws Exception {
+                                        if (!future.isSuccess()) {
+                                            channelPool.closeAndRelease(channel);
+                                            emitErrorIfSubscribed(future.cause());
+                                        }
                                     }
-                                }
-                            });
+                                });
+                            } catch (Exception e) {
+                                emitErrorIfSubscribed(e);
+                            }
 
                             if (request.body() == null) {
```

Comment:
Is this section still valid if an exception is caught on lines 277 - 278?

Reply:
No I don't think so-- actually, it might be easiest to just throw here. The reason these try/catch blocks were added is that addListener will fail after the EventLoopGroup is shut down.

Reply:
Actually, if I throw in this scope, Netty will just log an error and the Single will never complete. So I might just make a top-level try-catch for everything inside the channel acquire callback.